cli/command/secret/create.go

```diff
@@ -46,10 +46,6 @@ func runSecretCreate(dockerCli *command.DockerCli, options createOptions) error
 	client := dockerCli.Client()
 	ctx := context.Background()
 
-	if options.file == "" {
-		return fmt.Errorf("Please specify either a file name or STDIN ('-') with --file")
```

Comment:
IMHO if the user types `docker secret create foo ""`, the above is acceptable behavior.